src/workerd/jsg/jsg-test.c++

```diff
@@ -360,10 +360,48 @@ KJ_TEST("Test JSG_CALLABLE") {
   e.expectEval("let obj = getCallable(); new obj();", "boolean", "true");
 }
 
-}  // namespace
-
 // ========================================================================================
+struct InterceptContext: public ContextGlobalObject {
+  struct ProxyImpl: public jsg::Object,
+                    public jsg::NamedIntercept {
+    static jsg::Ref<ProxyImpl> constructor() { return jsg::alloc<ProxyImpl>(); }
+
+    int getBar() { return 123; }
+
+    // NamedIntercept implementation
+    kj::Maybe<jsg::JsValue> getNamed(jsg::Lock& js, kj::StringPtr name) override {
+      if (name == "foo") {
+        return kj::Maybe(js.str("bar"_kj));
+      }
+      return kj::none;
+    }
+
+    kj::Array<kj::String> listNamed(Lock& js) override {
+      return kj::arr(kj::str("foo"));
+    }
+
+    JSG_RESOURCE_TYPE(ProxyImpl) {
+      JSG_READONLY_PROTOTYPE_PROPERTY(bar, getBar);
+      JSG_NAMED_INTERCEPT();
+    }
+  };
 
+  JSG_RESOURCE_TYPE(InterceptContext) {
+    JSG_NESTED_TYPE(ProxyImpl);
+  }
+};
+JSG_DECLARE_ISOLATE_TYPE(InterceptIsolate, InterceptContext, InterceptContext::ProxyImpl);
+
+KJ_TEST("Named interceptor") {
+  Evaluator<InterceptContext, InterceptIsolate> e(v8System);
+  e.expectEval("p = new ProxyImpl; Object.keys(p); p.foo = 123; p.foo", "string", "bar");
+  e.expectEval("p = new ProxyImpl; p.bar", "number", "123");
+  e.expectEval("p = new ProxyImpl; Reflect.has(p, 'foo')", "boolean", "true");
```

Comment:
What gives this object the property that its fields are the same even when a new instance of it is constructed? Is it a side-effect of `JSG_NAMED_INTERCEPT`?

Reply:
Not sure what you mean here. The `ProxyImpl` is written to return specific values for both `bar` and `foo`.

Reply:
Oh I see. I thought that `Reflect.has(p, 'foo')` returning `true` meant that the `foo` set via `p.foo = 123` above was being stored across the different `ProxyImpl` instances. I didn't notice that `foo` was listed in `listNamed`.

Makes sense now.